js/script.js

```diff
@@ -0,0 +1,222 @@
+// ------------------------
+// LOGIC
+// ------------------------
+
+const Board = (() => {
+  let cells = new Array(9).fill(" ");
+
+  const fillCell = (cellNumber, cellSymbol) => {
+    cells[cellNumber] = cellSymbol;
+  }
+
+  const isCellTaken = (cellNumber) => {
+    return cells[cellNumber] != " ";
+  }
+
+  const resetBoard = () => {
+    cells.forEach((element, index) => cells[index] = " ");
```

Comment:
Hi @zenott
I tried before to reset the cells by using `cells = new Array(9).fill(" ");` but it doesn't update the cells and breaks the app. What goes wrong?

Reply:
Hi @mosaaleb ,
That is because this way you reassign the cells array which breaks the reference to the exported cells variable from the module. They are not the same we just pass one by reference to the other.
Instead you should just mutate the original array like
```
cells.fill(' ')
```
Hope it's clear.

Reply:
Thanks @Zenott for your response, that's make perfect sense! .But I am curious what if I want to update  i.e. single  integer variable, then should I make a getter for this variable to use it outside the module?

Reply:
Yes @mosaaleb , a getter function would get the updated value.

Reply:
Thanks @zenott